### PR TITLE
Detail actvity Refactoring, UI Refactoring

### PR DESCRIPTION
**Pull request by CebotariM**

### CODE REVIEW

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        //Intent testIntent = new Intent(this, HistoryActivity.class);
+        //startActivity(testIntent);
+
+
+
+
+
+
+
+
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java]: could you tell me please what this space means?

File: app/src/main/res/layout/weather_history_layout.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <com.astuetz.PagerSlidingTabStrip
+        android:id="@+id/tabs"
+        app:pstsShouldExpand="true"
+        app:pstsTextAllCaps="true"
+        android:layout_width="match_parent"
+        android:layout_height="48dp" />
```
Comment (MaximCeban) [inline app/src/main/res/layout/weather_history_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_item_layout.xml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        android:id="@+id/temp_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+android:layout_alignParentEnd="true"/>
+
+    <TextView
+        android:paddingTop="10dp"
+        android:layout_below="@id/image_view"
+android:layout_centerHorizontal="true"
+        android:textSize="16sp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_item_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_item_layout.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        android:textSize="20sp"
+        android:text="temp"
+        android:layout_alignTop="@id/image_view"
+        android:id="@+id/temp_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+android:layout_alignParentEnd="true"/>
+
+    <TextView
+        android:paddingTop="10dp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_item_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_item_layout.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        android:text="humidity"
+        android:id="@+id/humidity_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentEnd="true"
+        android:layout_alignBottom="@id/image_view"
+        android:paddingBottom="10dp"/>
+
+    <TextView
+        android:paddingTop="10dp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_item_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_item_layout.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        android:textSize="20sp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_item_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_item_layout.xml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    <TextView
+        android:textSize="18sp"
+        android:text="humidity"
+        android:id="@+id/humidity_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentEnd="true"
+        android:layout_alignBottom="@id/image_view"
+        android:paddingBottom="10dp"/>
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_item_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_item_layout.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    <TextView
+        tools:background="@drawable/rain_day"
+        android:textSize="72sp"
+        android:id="@+id/image_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"/>
+
+    <TextView
+        android:textSize="18sp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_item_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_item_layout.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:padding="?attr/listPreferredItemPaddingLeft"
+    android:background="@color/white">
+
+    <TextView
+        tools:background="@drawable/rain_day"
+        android:textSize="72sp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_item_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_history_layout.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<GridView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/weather_grid_view"
+    android:numColumns="2"
+    android:gravity="center"
+    android:background="@color/colorPrimaryDark"
+    android:columnWidth="50dp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_history_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_history_layout.xml [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    android:horizontalSpacing="5dp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_history_layout.xml]: hardcored value

File: app/src/main/res/layout/grid_history_layout.xml [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+    android:verticalSpacing="5dp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/grid_history_layout.xml]: hardcored value

File: app/src/main/res/layout/detail_today_forecast_recycler_row.xml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        android:textSize="@dimen/icon_medium"
+        android:layout_alignParentTop="true"
+        android:layout_alignParentStart="true"
+        android:gravity="center"/>
+    <TextView
+        android:id="@+id/detail_today_temperature"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="temperature"
+        android:textSize="27sp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/detail_today_forecast_recycler_row.xml]: hardcored value

File: app/src/main/res/layout/detail_today_forecast_recycler_row.xml [hunk 14]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="wrap_content"
+    android:layout_height="match_parent"
+    android:layout_marginEnd="10dp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/detail_today_forecast_recycler_row.xml]: hardcored value

File: app/src/main/res/layout/detail_today_forecast_recycler_row.xml [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    android:layout_marginStart="10dp">
```
Comment (MaximCeban) [inline app/src/main/res/layout/detail_today_forecast_recycler_row.xml]: hardcored value

File: app/src/main/res/layout/detail_parameters_card.xml [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+        <TextView
+            android:id="@+id/detail_parameter_card_title"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/detail_parameter_card_title"
+            android:layout_margin="@dimen/standart_margin"
+            android:textSize="@dimen/card_title_size"
+            android:textStyle="italic" />
+
+        <include layout="@layout/break_line" />
```
Comment (MaximCeban) [inline app/src/main/res/layout/detail_parameters_card.xml]: https://developer.android.com/reference/android/widget/LinearLayout.html#attr_android:divider

File: app/src/main/res/layout/detail_main_layout.xml [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+
+        <android.support.design.widget.CollapsingToolbarLayout
+            android:id="@+id/detail_main_collapsing_toolbar"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:fitsSystemWindows="true"
+            app:contentScrim="@drawable/gradient_background"
+            app:expandedTitleMarginStart="@dimen/detail_city_name_margin"
+            app:layout_scrollFlags="scroll|exitUntilCollapsed">
+            <View
```
Comment (MaximCeban) [inline app/src/main/res/layout/detail_main_layout.xml]: could you tell me please what this view does?

File: app/src/main/res/layout/detail_main_layout.xml [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"/>
+            <android.support.v7.widget.Toolbar
+                android:id="@+id/detail_main_toolbar"
+                android:layout_width="match_parent"
+                android:layout_height="?attr/actionBarSize"
+                android:background="@drawable/gradient_background"
+                android:textAlignment="center"
+                app:layout_collapseMode="pin" />
+
```
Comment (MaximCeban) [inline app/src/main/res/layout/detail_main_layout.xml:34]: try to avoid empty spaces

File: app/src/main/res/layout/detail_forecast_card.xml [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="start"
+            android:layout_marginStart="@dimen/standart_margin"
+            android:text="@string/forecast_card_title"
+            android:textSize="@dimen/card_title_size"
+            android:textStyle="italic" />
+
+        <include layout="@layout/break_line" />
```
Comment (MaximCeban) [inline app/src/main/res/layout/detail_forecast_card.xml]: https://developer.android.com/reference/android/widget/LinearLayout.html#attr_android:divider

File: app/src/main/res/layout/break_line.xml [hunk 20]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<View
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="1dp"
```
Comment (MaximCeban) [inline app/src/main/res/layout/break_line.xml]: hardcored value

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/weather/forecast/adapter/ForecastRecyclerViewAdapter.java [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+        this.cityForecastList = cityForecastList;
+        this.listener = listener;
+        sharedPrefs = new SharedPrefs(context);
+        imageUtils = new ImageUtils(context);
+    }
+
+    @Override
+    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.forecast_recycler_row, parent, false);
+        return new CustomViewHolder(view);
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/weather/forecast/adapter/ForecastRecyclerViewAdapter.java]: could you tell me please what _CustomViewHolder_ means? Do you remember about meaningful naming?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/weather/forecast/adapter/ForecastRecyclerViewAdapter.java [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+        public void bind(final CityForecast cityForecast) {
+            final WeatherData currentWeather = cityForecast.getCurrentCityWeather();
+            String weatherIconId = currentWeather.getWeatherIcon();
+            cityName.setText(cityForecast.getCityName());
+            cityWeatherDescription.setText(currentWeather.getWeatherDescription());
+
+            int kelvinTemperature = cityForecast.getCurrentCityWeather().getTemp().intValue();
+
+            cityTemperature.setText(TemperatureConvertor.fromId(sharedPrefs.getTempScaleFromPrefs()).convertToTemperature(kelvinTemperature));
+            cityID = cityForecast.getCityId();
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/weather/forecast/adapter/ForecastRecyclerViewAdapter.java]: for what do we need it?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        ButterKnife.bind(this);
+
+
+        MainPresenter mainPresenter = new MainPresenter(this, this);
+
+        mainPresenter.checkLastUpdateTime();
+        Bundle bundle = new Bundle();
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java]: for what this variable was  created?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+    @BindView(R.id.forecast_progress_bar)
+    ProgressBar progressBar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        ButterKnife.bind(this);
+
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java]: empty space for what do  you add it?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/adapter/GridHistoryAdapter.java [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+import com.example.nvdovin.weatherapp.data.model.WeatherData;
+import com.example.nvdovin.weatherapp.domain.utils.time.TimeUtils;
+import com.example.nvdovin.weatherapp.domain.utils.design.ImageUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GridHistoryAdapter extends BaseAdapter {
+
+    private static final String HOUR_PRECISION = ":00";
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/adapter/GridHistoryAdapter.java]: hour_precission? really? so, j could you explain please why do you named it in this manner?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/GridHistoryPresenter.java [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+
+import com.example.nvdovin.weatherapp.data.model.WeatherData;
+import com.example.nvdovin.weatherapp.domain.factory.GreenDaoFactory;
+import com.example.nvdovin.weatherapp.domain.utils.time.TimeUtils;
+
+import java.util.List;
+
+public class GridHistoryPresenter {
+
+    private static final Long DAY_IN_MILLISECONDS_EXCLUSIVE = 75600L;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/GridHistoryPresenter.java]: Could you tell me please what _EXCLUSIVE means? what kind of a day it is?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/GridHistoryFragment.java [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class GridHistoryFragment extends Fragment implements GridHistoryView {
+
+    public static final String DAY_TIMESTAMP = "DAY_TIMESTAMP";
+    public static final String CITY_ID = "CITY_ID";
+    private GridView weatherGridView;
+    private GridHistoryAdapter gridHistoryAdapter;
+    private List<WeatherData> weatherDataList;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/GridHistoryFragment.java]: could you explain please for what  you created this global variable?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/adapter/HistoryViewPagerAdapter.java [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+    private Long cityId;
+
+    public HistoryViewPagerAdapter(FragmentManager fm) {
+        super(fm);
+        timestampList = new ArrayList<>();
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        return GridHistoryFragment.newInstance(timestampList.get(position), cityId);
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/adapter/HistoryViewPagerAdapter.java]: pay attention on http://docs.oracle.com/javase/7/docs/api/java/lang/ArrayIndexOutOfBoundsException.html

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryView.java [hunk 29]
```diff
@@ -0,0 +1,7 @@
+package com.example.nvdovin.weatherapp.presentation.history;
+
+public class HistoryView {
+
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryView.java]: empty class?????? for what?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryPresenter.java [hunk 30]
```diff
@@ -0,0 +1,7 @@
+package com.example.nvdovin.weatherapp.presentation.history;
+
+public class HistoryPresenter {
+
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryPresenter.java]: another empty class?????

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryActivity.java [hunk 31]
```diff
@@ -0,0 +1,65 @@
...
+    private void setupViewPager(ViewPager viewPager) {
+        Long[] tmpDates = timeUtils.getDateArrayByDate(Calendar.getInstance(TimeZone.getTimeZone("UTC")), DAYS_AGO, DAYS_AHEAD);
+        HistoryViewPagerAdapter historyViewPagerAdapter = new HistoryViewPagerAdapter(getSupportFragmentManager());
+        historyViewPagerAdapter.addCity(cityId);//TODO implement cityId logic
+        for (int i = 0; i < tmpDates.length; i++) {
+            historyViewPagerAdapter.addDay(tmpDates[i]);
+        }
+        viewPager.setAdapter(historyViewPagerAdapter);
+        viewPager.setCurrentItem(DAYS_AGO);
+        viewPager.setOffscreenPageLimit(5);
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryActivity.java]: 5? what does it means?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryActivity.java [hunk 32]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_history);
+
+        ButterKnife.bind(this);
+
+        cityId = getIntent().getLongExtra(TIMESTAMP_KEY, DEFAULT_VALUE);
+        Toast.makeText(this, "" + cityId, Toast.LENGTH_SHORT).show();
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/HistoryActivity.java]: String.valueOf(int a) not "" + int

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/TodayForecastAdapter.java [hunk 33]
```diff
@@ -0,0 +1,74 @@
...
+        );
+        holder.time.setText(dateConvertor.convertLongToFormat(weatherData.getDt(), DATE_FORMAT));
+    }
+
+    @Override
+    public int getItemCount() {
+        return dailyForecast.getWeatherDataList().size();
+    }
+
+    class CustomViewHolder extends RecyclerView.ViewHolder {
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/TodayForecastAdapter.java]: CustomViewHolder??????? meaningful name, just  takes a minute for  thinking  about what it is related for, and then  name it in the proper way.

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/NextDaysRecyclerAdapter.java [hunk 34]
```diff
@@ -0,0 +1,79 @@
...
+                                .convertToTemperature(dailyForecast.getDayTempMax()))
+        );
+    }
+
+    @Override
+    public int getItemCount() {
+        return NUMBER_OF_DAYS_TO_FORECAST;
+    }
+
+    class CustomViewHolder extends RecyclerView.ViewHolder {
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/NextDaysRecyclerAdapter.java]: CustomViewHolder??????? meaningful name, just  takes a minute for  thinking  about what it is related for, and then  name it in the proper way.

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/NextDaysRecyclerAdapter.java [hunk 35]
```diff
@@ -0,0 +1,79 @@
...
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.detail_five_days_recycler_row, parent, false);
+        return new NextDaysRecyclerAdapter.CustomViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(CustomViewHolder holder, int position) {
+        DailyForecast dailyForecast = dailyForecastList.get(position);
+        if (position == 0) {
+            holder.name.setText(R.string.today);
+        } else if (position == 1) {
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/NextDaysRecyclerAdapter.java]: what position ==1 means? what _1_ means?

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/MainRecyclerAdapter.java [hunk 36]
```diff
@@ -0,0 +1,134 @@
...
+        TextView windspeed;
+
+        ParametersCard(View itemView) {
+            super(itemView);
+            imageUtils.setTypeface(icon);
+        }
+
+        @Override
+        public void bindData(DailyForecast data) {
+
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/MainRecyclerAdapter.java]: empty line remove it!

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/MainRecyclerAdapter.java [hunk 37]
```diff
@@ -0,0 +1,134 @@
...
+                    (context, LinearLayoutManager.HORIZONTAL, false));
+            todayDataRecycler.setAdapter(new TodayForecastAdapter(dailyForecast, context));
+
+            nextDaysRecycler.setLayoutManager(new LinearLayoutManager(context));
+            nextDaysRecycler.setNestedScrollingEnabled(false);
+            nextDaysRecycler.setAdapter(new NextDaysRecyclerAdapter(dailyForecastList, context));
+        }
+    }
+
+    class ParametersCard extends ViewHolder<DailyForecast> {
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/MainRecyclerAdapter.java]: I guess a better name for that will be ParametersCardHolder if it represents a specific kind of ViewHolder.

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/DetailActivity.java [hunk 38]
```diff
@@ -0,0 +1,83 @@
...
+        Intent intent = getIntent();
+        Bundle bundle = intent.getBundleExtra(DETAIL_BUNDLE);
+        Long cityId = bundle.getLong(CITY_ID_KEY);
+        Long timestamp = bundle.getLong(TIMESTAMP_KEY);
+
+        DetailsPresenter detailsPresenter = new DetailsPresenter(this);
+
+        toolbar.setTitle(detailsPresenter.getCityName(cityId));
+
+        toolbar.setNavigationIcon(R.drawable.ic_arrow_back_white_24dp);
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/DetailActivity.java]: 24dp does it helps you?

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/time/TimeUtils.java [hunk 39]
```diff
@@ -0,0 +1,63 @@
...
+    private static final int RESET_VALUE = 0;
+    private static final int PERIOD = 3;
+    public static final int COUNT_OF_RECORDINGS_FOR_DAY = 8;
+
+    public Long[] getAllPeriodsForDay(Long timestamp) {
+        Long dates[] = new Long[COUNT_OF_RECORDINGS_FOR_DAY];
+        Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        calendar.setTimeInMillis(timestamp * MILLISECONDS);
+        calendar.set(Calendar.HOUR_OF_DAY, RESET_VALUE);
+        /*calendar.set(Calendar.MINUTE, RESET_VALUE);
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/time/TimeUtils.java]: why do you push commented code?

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/ImageUtils.java [hunk 40]
```diff
@@ -0,0 +1,38 @@
...
+        this.context = context;
+        weatherFont = Typeface.createFromAsset(context.getAssets(), FONTS_LOCATION);
+    }
+
+    public int setWeatherIcon(String id, TextView iconTextView) {
+        WeatherCodesMap weatherCodesMap = new WeatherCodesMap();
+        int resID = weatherCodesMap.getBackgroundResById(id);
+        String icon = context.getString(weatherCodesMap.getIconByID(id));
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N)
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/ImageUtils.java]: use braces

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/DateConvertor.java [hunk 41]
```diff
@@ -0,0 +1,29 @@
+package com.example.nvdovin.weatherapp.domain.utils;
+
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateConvertor {
+
+    public static final int MILI_VALUE = 1000;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/DateConvertor.java]: MILI_ whaaaat?

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/factory/GreenDaoFactory.java [hunk 42]
```diff
@@ -0,0 +1,149 @@
...
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class GreenDaoFactory {
+
+    private static final String DB_NAME = "WEATHER_DB";
+    private static final Long DAY_IN_MILLISECONDS_INCLUSIVE = 86400L;
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/factory/GreenDaoFactory.java]: it is not funny what a hell with inclusive and exclusive

File: app/build.gradle [hunk 43]
```diff
@@ -41,4 +41,9 @@ dependencies {
     compile 'com.squareup.okhttp3:logging-interceptor:3.4.1'
     compile 'org.greenrobot:greendao:3.2.0'
     compile 'org.greenrobot:eventbus:3.0.0'
+    compile 'com.astuetz:pagerslidingtabstrip:1.0.1'
```
Comment (MaximCeban) [inline app/build.gradle]: remove unused libraries

**Review by MaximCeban — DISMISSED** [review]

**Comment by MaximCeban** [comment]
Man, please try to avoid pushing huge commits, please, here is the bunch of files

File: app/build.gradle (hunk 43 above)
Comment (CebotariM) [reply app/build.gradle]: The library is needed for the History Activity

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/DateConvertor.java (hunk 41 above)
Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/DateConvertor.java]: The constant needed to convert from Milliseconds to seconds, had to extract 1000 in to constant

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/GridHistoryPresenter.java (hunk 26 above)
Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/GridHistoryPresenter.java]: INCLUSIVE means 21 hours, or almost a day. if we put 86400 sec (EXCLUSIVE) we get the first data for the next day also. Say it is 00.00.
INCLUSIVE - we get the weather data until 21.00
EXCLUSIVE - we get the data until tomorrow at 00.00, which is the next day

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java (hunk 24 above)
Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/presentation/main/MainActivity.java]: *Fixed

File: app/src/main/res/layout/detail_main_layout.xml (hunk 18 above)
Comment (CebotariM) [reply app/src/main/res/layout/detail_main_layout.xml:34]: *Fixed

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/ImageUtils.java (hunk 40 above)
Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/ImageUtils.java]: *Fixed

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/temperature/TemperatureConvertor.java [hunk 44]
```diff
@@ -0,0 +1,55 @@
...
+            String temperature = String.valueOf(kelvinValue - 273);
+            SpannableString temperatureWithSymbol=  new SpannableString(temperature.concat(TemperatureConvertorInterface.CELSIUS_SIGN));
+            temperatureWithSymbol.setSpan(new RelativeSizeSpan(TemperatureConvertorInterface.PROPORTION), 0, temperature.length() , 0);
+            return temperatureWithSymbol;
+        }
+    },
+    FAHRENHEIT(1) {
+        @Override
+        public SpannableString convertToTemperature(int kelvinValue) {
+            String temperature = String.valueOf(kelvinValue * 9 / 5 - 459);
```
Comment (victorjimbei) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/temperature/TemperatureConvertor.java]: Oh this values... I remember we agreed to have them declared in interface

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/date/DateConvertor.java [hunk 45]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public Long convertDateToLong (Date date)
+    {
+        return date.getTime();
+    }
+
+    public Date convertLongToDate (Long dt)
+    {
+        return new Date(dt*1000);
```
Comment (victorjimbei) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/date/DateConvertor.java]: I mentioned this in some review before, this 1000 value doesn't look good here

File: app/src/main/java/com/example/nvdovin/weatherapp/data/network/response/GetCityListResponse.java [hunk 46]
```diff
@@ -11,6 +11,9 @@
     @Expose
     List<String> cities;
```
Comment (victorjimbei) [inline app/src/main/java/com/example/nvdovin/weatherapp/data/network/response/GetCityListResponse.java]: Oh this spaces... It's all about code formatting

File: app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/ImageUtils.java [hunk 47]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            iconTextView.setText(Html.fromHtml(icon, Html.FROM_HTML_MODE_LEGACY));
+        }
+
+        else
```
Comment (victorjimbei) [inline app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/design/ImageUtils.java]: Code formatting... looks like this else is missed in space and has nothing with if above

File: app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/NextDaysRecyclerAdapter.java [hunk 48]
```diff
@@ -0,0 +1,103 @@
...
+            ButterKnife.bind(this, itemView);
+        }
+    }
+
+    public String getDayString(Date dateToCheck){
+        String dayString = "";
+        Date todayDate = TimeUtils.getTodayDate();
+
+        if (TimeUtils.isSameDate(dateToCheck, TimeUtils.addDaysToDate(todayDate, YESTERDAY_SUBSTRACT_DAY)))
+            {
```
Comment (victorjimbei) [inline app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/NextDaysRecyclerAdapter.java]: please change code style for if/else to have the brakets in the same line
if(){
}else{
}

**Review by victorjimbei — APPROVED** [review]

**Pull request merged**